### DISCUSSION TITLE
Assign different material to each chain for fbx export?

### DISCUSSION THREAD

**Discussion by beka89UE** [BradyAJohnston/MolecularNodes#379] [Q&A]
How can I export as an fbx with color data? I've used color by chain. When I export as a fbx to C4D it just imports with a white material. Is there a way to assign a different material to each chain? I'd like to change the material of each chain using Redshift materials in C4D.

Thanks!

**Comment by BradyAJohnston** [comment]
In the setup that I use, each of the vertices get an attribute called `Color` which has the color information that is used inside of a single shader to change the colors. I am unfamiliar with Redshift, or whether the attributes will be accessible in redshift materials / exported with FBX. 

Otherwise the easiest approach would be to apply the modifier, split the chains into separate objects, then apply a different material to each. 

I could maybe add a node that lets you assign a material for each chain which might also make it easier.

**Reply by beka89UE** [reply]
Other 3D programs like C4D and Maya can't use those color attributes from my testing, unfortunately.

How do you split the chains into separate objects in blender? I'm more familiar with C4D and Maya.

An additional node would be incredibly helpful!

**Reply by BradyAJohnston** [reply]
I'll look into creating a node around different materials. I haven't really done much exporting, so any other requests around what would be good for improving the exporting workflow, let me know.

**Reply by BradyAJohnston** [reply]
If you apply the Molecular Nodes modifier, you can separate the model into the different chains with `Tab` (edit mode) -> `P` (separate) -> By loose parts.

You can then alter the materials on each individual object, before exporting them.

https://github.com/BradyAJohnston/MolecularNodes/assets/36021261/5e27da70-e49e-4bc1-ab08-32f6ff998b80
